improved compatibility with HIRO-7

sgritsenko opened the pull request:
Adding compatibility with HIRO-7 and the latest version of Decision Service.

sgritsenko committed:
improved compatibility with HIRO-7

Adding compatibility with HIRO-7 and the latest version of Decision Service.

SteffenBauer commented on python/hiro-client.py:
```diff
@@ -42,16 +43,25 @@ def __init__(self, username, password,
         if token is not None:
             headers['Authorization'] = "Bearer " + token
         res = requests.get(url, headers=headers, verify=False)
-        return self._parse_response(res)
```
Why reverting the handling of the requests response here, from a single method `_parse_response` to individual code blocks for get, post, delete? It looks to me like the handling block ( `if res.status_code == 401 ...return res.text`) is the same for the three access modes get/post/delete, so we have code duplication here.

SteffenBauer reviewed (approved):
Principally okay, just one minor issue with code style (duplicated code which should be moved to a separate method)

sgritsenko committed:
Update hiro-client.py

mglusiuk commented on python/hiro-client.py:
```diff
@@ -42,6 +43,8 @@ def __init__(self, username, password,
 
     @backoff.on_exception(*BACKOFF_ARGS, **BACKOFF_KWARGS)
     def get(self, url, token=None):
+        url = url.replace('"','') # temporary workaround.
```
could you pls explain where " comes from?

sgritsenko replied:
That's a bit weird, but for example in my IDE if i don't remove ", then it throws an exception from requests library.
At the same time I can successfully run this code directly on the server via console.
This was done mostly with a purpose of compatibility and I plan to improve it the future (re-write how this URL is forming: url = self._graph_endpoint + '/' + quote_plus(node_id))

<img width="1414" alt="Screenshot 2020-07-29 at 10 20 20" src="https://user-images.githubusercontent.com/36957194/88776714-9989f700-d186-11ea-86bf-acbf28b51c04.png">

mglusiuk replied:
maybe you initialize it with quoted graph_endpoint instead of plain string url

sgritsenko committed:
Update hiro-client.py

removed temporary workaround.
it really was the problem with how environment variables are defined in my IDE.
before it was this: LS_ENV_GRAPH_ENDPOINT="https://someurl"
now i changed it to this: LS_ENV_GRAPH_ENDPOINT=https://someurl
and this fixed this issue.
thanks Miroslaw to pointing me to the right direction.
